garden-service/src/plugins/kubernetes/hot-reload.ts

```
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2018 Garden Technologies, Inc. <info@garden.io>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as Bluebird from "bluebird"
+import * as execa from "execa"
+import { waitForServices } from "./status"
+import { HotReloadParams } from "../../types/plugin/params"
+import { ContainerModule } from "../container"
+import { HotReloadResult } from "../../types/plugin/outputs"
+import { getAppNamespace } from "./namespace"
+import { RSYNC_PORT } from "./service"
+import { rsyncSourcePath, rsyncTargetPath } from "./deployment"
+import { kubectl } from "./kubectl"
+import getPort = require("get-port")
+import { RuntimeError } from "../../exceptions"
+
+export async function hotReload(
+  { ctx, log, runtimeContext, module, buildDependencies }: HotReloadParams<ContainerModule>,
+): Promise<HotReloadResult> {
+  const hotReloadConfig = module.spec.hotReload!
+  const services = module.services
+
+  if (!await waitForServices(ctx, log, runtimeContext, services, buildDependencies)) {
+    // Service deployment timed out, skip hot reload
+    return {}
```

This feels bad to just ignore errors? can't we throw an exception or make it more visible it didn't work?

---

This is not new in the PR (see the old code), maybe @thsig can shed light on this.

---

But if I understand correctly, the user will see an error bubble up elsewhere, and an error thrown here would be superfluous.

---

We should at the least log it...

This behaviour is the exact same as the success result of returning an empty hash.

---

Looking back at this, I think we should throw an exception here (or in `waitForServices`) if a timeout occurs.

I think I wrote something of the sort at the time, but it seems to have ended up being omitted when I was refactoring/pruning the original hot reload branch at the time.